README.md

```diff
@@ -109,6 +109,15 @@ default values only after having loaded this script into your ZSH session.
   causes this script to perform a case-insensitive search. See the "Globbing
   Flags" section in the zshexpn(1) man page to learn about the kinds of
   values you may assign to this variable.
+  
+* `HISTORY_SUBSTRING_SEARCH_METHOD` is a global variable that defines
+  how the command history will be searched for your query.
+  * 'default' - find history by string
+    * e.g. 'a b c' will match 'xxxA B Cxxx'
+  * 'fuzzy_words' - find history by fuzzy word matching
+    * e.g. 'ab c' will match 'xxxABxxxCxxx'
+  * 'fuzzy_chars' - find history by fuzzy character matching
+    * e.g. 'abc' will match 'xxxAxxxBxxxCxxx'
 
```

Comment:
This description is inaccurate:
- If you set the variable to anything other than `fuzzy_words` or `fuzzy_chars` (including the misspelled versions of these, you get the default behaviour. Considering my general comment on my doubt about the `fuzzy_chars` method, I would prefer to have a single binary option `HISTORY_SUBSTRING_SEARCH_FUZZY`, which if non empty, uses the current `fuzzy_words` method, and if it is empty, then it uses the default method.
- The matches you describe are only valid if `HISTORY_SUBSTRING_SEARCH_GLOBBING_FLAGS='i'` (i.e.: case insensitive match), which is only the default value. A better description is that `a b c` will match `*a b c*`, `ab c` will match `*ab*c*`, and `abc` will match `*a*b*c*`.
- If it is not obvious after the above change, then it is also worth pointing out that `foo bar` will match `*foo*bar*` but not `*bar*foo*`